tests/Primitives.test.ts

```diff
@@ -79,4 +80,14 @@ describe("Primitives", () => {
 
     expectTypeOf<actualPrimitives>().toEqualTypeOf<expectedPrimitives>();
   });
+
+  it("should infer the primitive properties from metadata", () => {
```

Comment:
I would ask you to please accept this suggestion to merge this PR:

```suggestion
  it("should infer properties with unknown type", () => {
```

Why: It would be better to express which are the particularities introduced by the `Metadata` class and leaving it as an implementation detail on how to test these particulatities.

That is, if I am a newcomer to the library and review the test suite to have an idea about what are the supported use cases, would I be able to infer them by the test case titles? If I am a maintainer and want to clean-up redundant test-cases or refactor this test suite, will I be able to understand the implications of removing this test case by just its name?